test/suite/test_cache_tracking01.py

```diff
@@ -0,0 +1,89 @@
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+from wtdataset import SimpleDataSet
+
+# Ensure cache is tracked correctly for page disk images.
+class test_cache_tracking(wttest.WiredTigerTestCase):
+    uri = "table:test_cache_tracking"
+    conn_config = "cache_size=1GB,statistics=(all),statistics_log=(wait=1,json=true,on_close=true)"
+
+    format_values = [
+        ('column', dict(key_format='r', value_format='S')),
+        ('row_integer', dict(key_format='i', value_format='S')),
+    ]
+    compressors = [
+        ('none', dict(compressor='none')),
+        ('snappy', dict(compressor='snappy')),
+    ]
+    scenarios = make_scenarios(format_values, compressors)
+
+    # Load the compressor extension, skip the test if missing.
+    def conn_extensions(self, extlist):
+        extlist.skip_if_missing = True
+        extlist.extension('compressors', self.compressor)
+
+    def get_stat(self, stat):
+        stat_cursor = self.session.open_cursor('statistics:')
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+
+    def test_cache_tracking(self):
+        _128MB = 128000000
+        nrows = 10
+        conf = 'allocation_size=128MB,leaf_page_max=128MB,internal_page_max=128MB,' \
+            'block_compressor={}'.format(self.compressor)
+        
+        ds = SimpleDataSet(
+            self, self.uri, nrows, key_format=self.key_format, value_format=self.value_format, config=conf)
+        ds.create()
+        cursor = ds.open_cursor(self.uri, None)
+
+        for i in range(1, nrows):
+            cursor[ds.key(i)] = ds.value(i)
+
+        # Evict pages to disk.
+        evict_cursor = self.session.open_cursor(self.uri, None, "debug=(release_evict)")
+        for i in range(1, nrows):
+            evict_cursor.set_key(ds.key(i))
+            self.assertEquals(evict_cursor.search(), 0)
+            evict_cursor.reset()
+
+        # Read a page back in memory.
+        cursor.set_key(ds.key(1))
+        cursor.search()
+
+        cache_bytes_page_image = self.get_stat(wiredtiger.stat.conn.cache_bytes_image)
+
+        if self.compressor == "none":
+            self.assertGreaterEqual(cache_bytes_page_image, _128MB)
+        else:
+            self.assertLess(cache_bytes_page_image, _128MB)
```

Comment:
I would also add a comment here explaining the logic here. It is easy to see what comparisons the test makes. But for someone not familiar with the issue, it probably won't be clear why the test asserts these things.

I.e., explain that when there is compression, we allocate pages the amount of memory needed for the in-memory page, but when we have a page that isn't compressed, we allocate space based on the on-disk size rather than the in-memory size. This test is set up, so that 

Also, I would expect that the `assertGreaterEqual` here could just be `assertGreater`. To read the key from disk, we'll load both the root page and the leaf page, so we should have at least two 128MB pages in memory.

Reply:
Updated in [9ace2ef](https://github.com/wiredtiger/wiredtiger/pull/9279/commits/9ace2effd9697c5bcf503a644f5829d49c8f58c2)